tests/conftest.py

```diff
@@ -21,3 +21,38 @@ def app(django_app):
 @pytest.fixture
 def csrf_exempt_django_app(django_app_factory):
     return django_app_factory(csrf_checks=False)
+
+
+@pytest.fixture(autouse=True)
+def _haystack_marker(request):
+    """
+    Implement the 'haystack' marker.
+
+    This rebuilds the index at the start of each test and clears it at the end.
+
+    Takes an optional connection parameter; if set, clears and rebuilds
+    only the specified haystack connections.
+    """
+    marker = request.keywords.get('haystack', None)
+
+    if marker:
+        from pytest_django.lazy_django import skip_if_no_django
+        from django.core.management import call_command
+        request.getfixturevalue('db')
+
+        # optional haystack connection parameter
+        # if specified, pass to clear_index and rebuild_index
+        connection = marker.kwargs.get('connection', None)
+        index_args = {}
+        if connection:
+            index_args['using'] = connection
```

Comment:
```suggestion
        index_args['using'] = 'test'
```

This way we will be able to get rid of `test_settings.py`, by simply adding another `'test'` connection alongside `'default'` as it is done with database configuration settings.

Reply:
I have tried this, but the problem is that settings are always loaded with 'default' value. Then when the tests are run, it starts to use 'test', but what happens is that 'default' and 'test' connections mix up and the search index has data from both development database and test database, which causes tests to fail and then we also need to rebuild the index  for development after running the tests, because it has test data.

Reply:
I think, in order to make this work, you need to specify a different connection in settings, here:

```python
HAYSTACK_CONNECTIONS = {
    'default': env.search_url(),
   'test': env.search_url(default='elasticsearch://localhsot:9200/test'),
}
```

Reply:
Yes, I tried it by adding 'test' connection to the settings, but result was as I said before.

Reply:
I fixed it for you.